src/utils.jl

```diff
@@ -49,6 +49,8 @@ Else return the one and only element.
 function getattr(o::Observable, idx, default=nothing)
     if o[] isa AbstractVector && !isa(o[], Point)
         return o[][idx]
+    elseif o[] isa DefaultDict || o[] isa DefaultOrderedDict
+        return getindex(o[], idx)
     elseif o[] isa AbstractDict
         return get(o[], idx, default)
```

Comment:
its a pity that `get(dict, key, default)` is not implemented for `DefaultDict` in a way that the "internal" default value is chosen over the "external" default defined in `get` call. Otherwise the `AbstractDict` codepath would just work

Reply:
```julia
julia> d = DefaultDict(:default, 1=>:foo, 2=>:bar)
DefaultDict{Int64, Symbol, Symbol} with 2 entries:
  2 => :bar
  1 => :foo

julia> get(d, 3, :wrongdefault)
:wrongdefault

julia> d[3]
:default

julia> get(d, 3, :wrongdefault)
:default
```
wat

Reply:
Yes, `DefaultDict` works like that; after you query a key that doesn't exist, it then stores the default value in it.
I think the reasoning behind is that a default value could be a function, and in this case you could you it for memoization purposes. There is also an example here https://juliacollections.github.io/DataStructures.jl/latest/default_dict/

Reply:
> its a pity that get(dict, key, default) is not implemented

Indeed a pity. that was also my perspective. Maybe we can start a discussion with the DataStructures.jl guys about another way `get(...)` could work but I wouldn't be too confident.